json/DataTypes/DecodeStudentLong.roc

```diff
@@ -0,0 +1,81 @@
+module [getListStudents]
+
+import JsonData exposing [Json]
+import Student exposing [Student, Module]
+
+JsonErrors : [
+    FieldNotFound Str,
+    ExpectedJsonObject Str,
+    ExpectedJsonArray Str,
+]
+
+getListStudents : Json -> Result (List Student) JsonErrors
+getListStudents = \json ->
+    when json is
+        Object obj ->
+            when Dict.get obj "students" is
+                Ok students -> checkStudentList students
+                _ -> Err (FieldNotFound "Expected field with name students in object")
```

Comment:
🎨 Stylistically, I 'd personally do this here:

```suggestion
                Err KeyNotFound -> Err (FieldNotFound "Expected field with name students in object")
```

This makes the pattern exhaustive, which is always my default preference so I get a compile error if I add a new tag and haven't handled it. That way I can decide on a case-by-case basis how I want to handle it, instead of the default being that the new tags silently go to the `_ ->` branch. 😄 

Granted, it's a safe bet that `Result` in particular isn't ever going to get any new tags, but I still like writing it out exhaustively in cases like this because now I can tell at a glance that all the possibilities are being handled. I also like that `Err KeyNotFound ->` in particular tells me when this branch will be taken: when the key isn't found!